src/analysis/processing/qgsalgorithmarrayfeatures.cpp

```diff
@@ -0,0 +1,221 @@
+/***************************************************************************
+                         qgsalgorithmarrayfeatures.cpp
+                         ---------------------
+    begin                : July 2018
+    copyright            : (C) 2018 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmarrayfeatures.h"
+
+///@cond PRIVATE
+
+QString QgsArrayFeaturesAlgorithm::name() const
+{
+  return QStringLiteral( "arrayfeatures" );
+}
+
+QString QgsArrayFeaturesAlgorithm::displayName() const
+{
+  return QObject::tr( "Array of offset features" );
+}
+
+QStringList QgsArrayFeaturesAlgorithm::tags() const
+{
+  return QObject::tr( "translate,duplicate,grid,spaced,moved,copy,features,objects,step,repeat" ).split( ',' );
+}
+
+QString QgsArrayFeaturesAlgorithm::group() const
+{
+  return QObject::tr( "Vector geometry" );
+}
+
+QString QgsArrayFeaturesAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeometry" );
+}
+
+QString QgsArrayFeaturesAlgorithm::outputName() const
+{
+  return QObject::tr( "Offset" );
```

Comment:
"Offset" is also the output name for the offset line algorithm's output. I'd suggest "Offsets Array", or maybe just add an "s"?